[CELEBORN-2186] Add correctness related section to bug report and PR templates (#3519)
apache/incubator-celeborn: .github/PULL_REQUEST_TEMPLATE.md

### Why are the changes needed?

> ### Is this PR solving a correctness bug?

Reviewer: ```suggestion
### Does this PR resolve a correctness bug?
```

> Yes/No. (If yes, please add `correctness` label to this Pull Request).

Reviewer: ```suggestion
<!-- Yes/No. Add `correctness` label for a correctness bugfix. -->
```

Author: @SteNicholas As we discussed, developers actually has no permission to set the label, how about:

```
Yes/No. (If yes, committer will add `correctness` label to this Pull Request).
```
